src/fsharp/vs/service.fs

```diff
@@ -1369,30 +1369,56 @@ type TypeCheckInfo
-              match cnr with 
-              // 'seq' in 'seq { ... }' gets colored as keywords
-              | CNR(_, (Item.Value vref), ItemOccurence.Use, _, _, _, m) when valRefEq g g.seq_vref vref -> 
-                  Some (m, FSharpTokenColorKind.Keyword) 
-              // custom builders, custom operations get colored as keywords
-              | CNR(_, (Item.CustomBuilder _ | Item.CustomOperation _), ItemOccurence.Use, _, _, _, m) -> 
-                  Some (m, FSharpTokenColorKind.Keyword) 
-              // types get colored as types when they occur in syntactic types or custom attributes
-              // typevariables get colored as types when they occur in syntactic types custom builders, custom operations get colored as keywords
-              | CNR(_, ( Item.TypeVar  _ 
-                       | Item.Types _ 
-                       | Item.UnqualifiedType _
-                       | Item.CtorGroup _), 
-                       ( ItemOccurence.UseInType 
-                       | ItemOccurence.UseInAttribute
-                       | ItemOccurence.Use _ 
-                       | ItemOccurence.Binding _
-                       | ItemOccurence.Pattern _), _, _, _, m) -> 
-                  Some (m, FSharpTokenColorKind.TypeName) 
-              | _ -> None)
-         |> Seq.toArray
+    member scope.GetExtraColorizations() =
+        sResolutions.CapturedNameResolutions
+        |> Seq.choose (fun cnr ->
+            match cnr with
+            // 'seq' in 'seq { ... }' gets colored as keywords
+            | CNR(_, (Item.Value vref), ItemOccurence.Use, _, _, _, m) when valRefEq g g.seq_vref vref ->
+                Some (m, FSharpTokenColorKind.Keyword)
+            // custom builders, custom operations get colored as keywords
+            | CNR(_, (Item.CustomBuilder _ | Item.CustomOperation _), ItemOccurence.Use, _, _, _, m) ->
+                Some (m, FSharpTokenColorKind.Keyword)
+            // well known type aliases get colored as keywords
+            | CNR(_, (Item.Types (n, _)), _, _, _, _, m) when TypeCheckInfo.IsKeywordType(n) ->
+                Some (m, FSharpTokenColorKind.Keyword)
+            // types get colored as types when they occur in syntactic types or custom attributes
+            // typevariables get colored as types when they occur in syntactic types custom builders, custom operations get colored as keywords
+            | CNR(_, ( Item.TypeVar _
+                     | Item.Types _
+                     | Item.UnqualifiedType _
+                     | Item.CtorGroup _),
+                     ( ItemOccurence.UseInType
+                     | ItemOccurence.UseInAttribute
+                     | ItemOccurence.Use _
+                     | ItemOccurence.Binding _
+                     | ItemOccurence.Pattern _), _, _, _, m) ->
+                Some (m, FSharpTokenColorKind.TypeName)
+            | _ -> None)
+        |> Seq.toArray
+
+    static member IsKeywordType(name : string) =
```

Comment:
It's better to use `keywordNames` https://github.com/Microsoft/visualfsharp/blob/6afc4f9776a5dba903ca6b11965a206fb8bc23d8/src/fsharp/lexhelp.fsi#L70. Probably convert it to a set and lookup.